core/src/main/java/com/scalar/db/storage/cassandra/MutateStatementHandler.java

```diff
@@ -46,27 +47,33 @@ public ResultSet handle(Operation operation) throws ExecutionException {
         throw new NoMutationException("No mutation was applied");
       }
       return results;
-
     } catch (WriteTimeoutException e) {
       logger.warn("Write timeout happened during mutate operation", e);
       if (e.getWriteType() == WriteType.CAS) {
         // retry needs to be done if applications need to do the operation exactly
-        throw new RetriableExecutionException("Paxos phase in CAS operation failed", e);
+        throw new RetriableExecutionException(
+            CoreError.CASSANDRA_WRITE_TIMEOUT_IN_PAXOS_PHASE_IN_MUTATION.buildMessage(), e);
```

Comment:
This error is in CONCURRENCY_ERROR category. But write timeout sounds to me it occurred since the Cassandra cluster was potentially busy not from internal transactional conflicts. So INTERNAL_ERROR might be safer. What do you think?

Reply:
We originally throw `RetriableExecutionException` that indicates an retryable error for this case. That's why I categorize it as `CONCURRENCY_ERROR`. What do you think?

Reply:
Thanks for sharing the background. It makes sense.

But WriteTimeoutException seemed to be likely thrown under heavy workload when googling it. So, my concern is still as follows:
1. Cassandra cluster is busy and a write operation is timed out
2. The client throws WriteTimeoutException to ScalarDB
3. ScalarDB wraps it with CONCURRENCY_ERROR
4. The application handles the wrapped exception and retries with a very small wait since it's CONCURRENCY_ERROR
5. The Cassandra cluster gets more busy...

IMO, INTERNAL_ERROR still fits this situation. But I understand the background, so this concern won't prevent this PR from being merged.

Reply:
Okay. At this time, this PR doesn't change the original behavior, we can continue this discussion in a separate PR or issue if needed. I'll merge this for now. Thank you for raising this discussion!